Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionResultPositiveOtherWarningFragment.kt

```diff
@@ -177,7 +187,31 @@ class SubmissionResultPositiveOtherWarningFragment : Fragment(),
     // InternalExposureNotificationPermissionHelper - callbacks
     override fun onKeySharePermissionGranted(keys: List<TemporaryExposureKey>) {
         super.onKeySharePermissionGranted(keys)
-        submissionViewModel.submitDiagnosisKeys(keys)
+        if (keys.isNotEmpty()) { submissionViewModel.submitDiagnosisKeys(keys)
+        } else { showSubmitErrorDialog()
+        }
+    }
+
+    /**
+     * Error Dialog
+     *
+     * If Diagnosis Key submission is failed, this error dialog is shown
+     * Cancellable is set to false
+     *
+     * @see SubmissionViewModel.submissionState
+     * @see AlertDialog.setCancelable
+     */
+    private fun showSubmitErrorDialog() {
```

Comment:
I´d suggest to use a more descriptive name for this. It´s not a general error dialog but specific for one special case

Reply:
refracted the name